src/context/authContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { AuthContextProps, AuthContextProviderProps, ClientApiUserResponse } from '@frontendTypes';
+import { RegisterRequest, UserResponse } from '@backendTypes';
+
+export const AuthContext = createContext<AuthContextProps | null>(null);
+
+export const AuthContextProvider = ({ children }: AuthContextProviderProps) => {
+  const [currentUser, setCurrentUser] = useState<UserResponse | null>(
+    JSON.parse(localStorage.getItem('user') || 'null'),
+  );
+
+  const login = async (inputs: RegisterRequest) => {
+    const data = await fetch('http://localhost:3001/api/auth/login', {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(inputs),
+    });
+    const response = (await data.json()) as ClientApiUserResponse;
+    if (response.ok) {
```

Comment:
Tutaj zastosowałbym try catch i nie wiem dlaczego po message znów jest message: 
 try {
    const response = await fetch(loginUrl, {
      method: 'POST',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(inputs),
    });

    if (!response.ok) {
      const errorResponse = await response.json();
      let errorMessage = 'Unknown error';

      if (Array.isArray(errorResponse?.errors)) {
        errorMessage = errorResponse.errors[0].message;
      } else if (errorResponse?.error?.message) {
        errorMessage = errorResponse.error.message;
      }

      throw new Error(errorMessage);
    }

    const responseData = await response.json();
    setCurrentUser(responseData?.data ?? null);
  } catch (error) {
    throw new Error(`Login failed: ${error.message}`);
  }

Reply:
to jest tylko funkcja którą przekazuje dalej do LoginPage gdzie jest:

`    try {
      await login(formValues);
      navigate('/dashboard');
    } catch (e) {
      alert(e);
    }`

co do błędu, to sam go tworze dzięki wiadomości z BE,
i go wyświetlam 

`throw new Error(
        Array.isArray(response.error)
          ? response.error[0].message.message
          : response.error?.message.message || 'Unknown error',
      );`

czemu message w message, bo tak dostaje z BE